components/script/bluetooth_blacklist.rs

```diff
@@ -39,74 +39,86 @@ pub enum ExcludeType {
 impl BluetoothBlacklist {
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted
-    pub fn is_excluded(&self, uuid: &str) -> bool {
-        self.0.is_empty() ||
-        self.0.get(uuid).map_or(false, |value| value.eq(&ExcludeType::Exclude))
+    pub fn is_blacklisted(&self, uuid: &str) -> bool {
+        match self.0 {
+            Ok(ref map) => map.get(uuid).map_or(false, |token| token.eq(&ExcludeType::Exclude)),
+            Err(_) => false,
+        }
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-reads
-    pub fn is_excluded_from_reads(&self, uuid: &str) -> bool {
-        self.is_excluded(uuid) ||
-        self.0.get(uuid).map_or(false, |value| value.eq(&ExcludeType::ExcludeReads))
+    pub fn is_blacklisted_for_reads(&self, uuid: &str) -> bool {
+        match self.0 {
+            Ok(ref map) => map.get(uuid).map_or(false, |t| t.eq(&ExcludeType::ExcludeReads) ||
+                                                           t.eq(&ExcludeType::Exclude)),
+            Err(_) => false,
+        }
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-writes
-    pub fn is_excluded_from_writes(&self, uuid: &str) -> bool {
-        self.is_excluded(uuid) ||
-        self.0.get(uuid).map_or(false, |value| value.eq(&ExcludeType::ExcludeWrites))
+    pub fn is_blacklisted_for_writes(&self, uuid: &str) -> bool {
+        match self.0 {
+            Ok(ref map) => map.get(uuid).map_or(false, |t| t.eq(&ExcludeType::ExcludeWrites) ||
+                                                           t.eq(&ExcludeType::Exclude)),
+            Err(_) => false,
+        }
     }
 }
 
 // https://webbluetoothcg.github.io/web-bluetooth/#parsing-the-blacklist
-fn parse_blacklist() -> HashMap<String, ExcludeType> {
-    // Step 3
-    let mut result = HashMap::new();
+fn parse_blacklist() -> Result<HashMap<String, ExcludeType>, ()> {
     // Step 1 missing, currently we parse ./resources/gatt_blacklist.txt.(zakorgy)
     let file = match File::open(FILE_PATH) {
         Ok(file) => file,
-        Err(_) => return result,
+        Err(_) => return Err(()),
     };
     let content = BufReader::new(file);
-    // Step 2 and 4
-    for line in content.lines() {
+    // Step 2
+    let lines = content.lines();
+    // Step 3
+    let mut result = HashMap::new();
+    // Step 4
+    for line in lines {
         if let Ok(line) = line {
             // Step 4.1
             if !line.is_empty() && !line.starts_with('#') {
-                let mut value = ExcludeType::Exclude;
+                let mut token = ExcludeType::Exclude;
```

Comment:
"following the spec" doesn't mean you have to use the same variable names :) `token` is used there because `"exclude"` seems to be a string, handled as a string token. Since we use enums, we don't have to call that as token.